src/main/java/com/iexec/worker/compute/ComputeManagerService.java

```diff
@@ -31,32 +31,54 @@
 import com.iexec.worker.docker.DockerService;
 import com.iexec.worker.result.ResultService;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 
 import java.io.File;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
 
 
 @Slf4j
 @Service
 public class ComputeManagerService {
 
     private static final String STDOUT_FILENAME = "stdout.txt";
+    /**
+     * Examples of timeout depending on category:
+     * <ul>
+     *     <li>XS:   50 * 0.02 = 1 minute</li>
+     *     <li>S :  200 * 0.02 = 4 minutes</li>
+     *     <li>M :  600 * 0.02 = 12 minutes</li>
+     *     <li>L : 1800 * 0.02 = 36 minutes</li>
+     *     <li>XL: 6000 * 0.02 = 120 minutes</li>
+     * </ul>
+     * If computed timeout duration is lower than {@link ComputeManagerService#minPullTimeout},
+     * then final timeout duration is {@link ComputeManagerService#minPullTimeout}.
+     * <br>
+     * If computed timeout duration is greater than {@link ComputeManagerService#maxPullTimeout},
+     * then final timeout duration is {@link ComputeManagerService#maxPullTimeout}.
+     */
+    private static final double PULL_TIMEOUT_FACTOR = 0.02;
 
     private final DockerService dockerService;
     private final PreComputeService preComputeService;
     private final AppComputeService appComputeService;
     private final PostComputeService postComputeService;
     private final WorkerConfigurationService workerConfigService;
     private final ResultService resultService;
+    @Value("${docker.image.min-pull-timeout:60}")
```

Comment:
```suggestion
    @Value("${docker.image-pull-timeout.min:60}")
```
So we have a shared base

Reply:
Good idea! See https://github.com/iExecBlockchainComputing/iexec-worker/pull/440/commits/debc550de31ebbcd8f178c6352270cac593b0759.

Reply:
Not sure this commit refers to comment (?)

Reply:
It includes it but it has a flaw (`docker.image.pull-timeout-max` instead of `docker.image.pull-timeout.max`). Let me fix it!